modules/rntuple.mjs

```diff
@@ -180,6 +178,14 @@ _readEnvelopeMetadata(reader) {
   return { envelopeType, envelopeLength };
 }
 
+_readSchemaExtension(reader) {
+  this._readFieldDescriptors(reader);
```

Comment:
Be careful: since in `_readFieldDescriptors` and `_readColumnDescriptors` you are doing `this.fieldDescriptors = fieldDescriptors` and similar, every time you call this function you will overwrite the previous ones.
You should either append to the existing array or return that array and handle it on the caller side.
Same goes for `_readAliasColumns` and `_readExtraTypeInformation`.

The Footer's schema description should be merged with the Header's to form the complete schema description (most of the time the Footer's schema extension will be empty btw).